napari/layers/labels/labels.py

```diff
@@ -340,6 +376,34 @@ def selected_label(self, selected_label):
         self._selected_color = self.get_color(selected_label)
         self.events.selected_label()
 
+    @property
+    def color_mode(self):
+        """Color mode string"""
+        return str(self._color_mode)
+
+    @color_mode.setter
+    def color_mode(self, color_mode: Union[str, LabelColorMode]):
+        color_mode = LabelColorMode(color_mode)
+        if color_mode == LabelColorMode.DIRECT:
+            (
+                custom_colormap,
+                label_color_index,
+            ) = colormaps.color_dict_to_colormap(self.color_dict)
+            self.colormap = custom_colormap
+            self._label_color_index = label_color_index
+        elif color_mode == LabelColorMode.AUTO:
+            self._label_color_index = {}
+            self.colormap = self._random_colormap
+        else:
+            raise ValueError("Unsupported Color Mode")
```

Comment:
This line will never be hit, because of `color_mode = LabelColorMode(color_mode)`, above. This will raise the following error if `color_mode` is not 'auto' or 'direct':

```pytb
~/conda/envs/pip7/lib/python3.7/enum.py in _missing_(cls, value)
    575     @classmethod
    576     def _missing_(cls, value):
--> 577         raise ValueError("%r is not a valid %s" % (value, cls.__name__))
    578
    579     def __repr__(self):

ValueError: 'hello' is not a valid LabelColorMode
```

So, the above two lines should be removed. In general, we should be paying more attention to our test coverage!

Reply:
This check also serves as future proof catchall trap. If we are to add another mode and this would help capture unexpected behavior during debugging. I found the catch under "Mode" useful when I implemented a new mode, so I am inheriting the same paradigm here as well

Reply:
@ziyangczi Ah, I see that rationale, which I hadn't appreciated before. Thanks. I think though that this is the wrong way to go about it. We shouldn't have inaccessible parts of the code in case they become accessible later. If you want an "early warning system" for unimplemented colormodes, I would suggest making a test parametrised on an iteration of all the possible modes in the enum — which will fail if you just let things fall through here.